include/ultra64/controller.h

```diff
@@ -63,155 +62,107 @@
 #define CONT_OVERRUN_ERROR              0x4
 
 /* Controller type */
-
 #define CONT_ABSOLUTE           0x0001
 #define CONT_RELATIVE           0x0002
 #define CONT_JOYPORT            0x0004
-#define CONT_EEPROM				0x8000
-#define CONT_EEP16K				0x4000
-#define	CONT_TYPE_MASK			0x1F07
-#define	CONT_TYPE_NORMAL		0x0005
-#define	CONT_TYPE_MOUSE			0x0002
-#define	CONT_TYPE_VOICE			0x0100
+#define CONT_EEPROM             0x8000
+#define CONT_EEP16K             0x4000
+#define CONT_TYPE_MASK          0x1F07
+#define CONT_TYPE_NORMAL        0x0005
+#define CONT_TYPE_MOUSE         0x0002
+#define CONT_TYPE_VOICE         0x0100
 
 /* Controller status */
-
 #define CONT_CARD_ON            0x01
 #define CONT_CARD_PULL          0x02
 #define CONT_ADDR_CRC_ER        0x04
-#define CONT_EEPROM_BUSY		0x80
+#define CONT_EEPROM_BUSY        0x80
 
 /* Buttons */
-
-#define CONT_A      0x8000
-#define CONT_B      0x4000
-#define CONT_G	    0x2000
-#define CONT_START  0x1000
-#define CONT_UP     0x0800
-#define CONT_DOWN   0x0400
-#define CONT_LEFT   0x0200
-#define CONT_RIGHT  0x0100
-#define CONT_L      0x0020
-#define CONT_R      0x0010
-#define CONT_E      0x0008
-#define CONT_D      0x0004
-#define CONT_C      0x0002
-#define CONT_F      0x0001
-
-/* Nintendo's official button names */
-
-#define A_BUTTON	CONT_A
-#define B_BUTTON	CONT_B
-#define L_TRIG		CONT_L
-#define R_TRIG		CONT_R
-#define Z_TRIG		CONT_G
-#define START_BUTTON	CONT_START
-#define U_JPAD		CONT_UP
-#define L_JPAD		CONT_LEFT
-#define R_JPAD		CONT_RIGHT
-#define D_JPAD		CONT_DOWN
-#define U_CBUTTONS	CONT_E
-#define L_CBUTTONS	CONT_C
-#define R_CBUTTONS	CONT_F
-#define D_CBUTTONS	CONT_D
-
-typedef struct
-{
-    /* 0x00 */ union{
-        u16 button;
-        struct{
-            u16 a   : 1;
-            u16 b   : 1;
-            u16 z   : 1;
-            u16 s   : 1;
-            u16 du  : 1;
-            u16 dd  : 1;
-            u16 dl  : 1;
-            u16 dr  : 1;
-            u16 illegal_h : 1;
-            u16 illegal_l : 1;
-            u16 l   : 1;
-            u16 r   : 1;
-            u16 cu  : 1;
-            u16 cd  : 1;
-            u16 cl  : 1;
-            u16 cr  : 1;
-        };
-    };
-    /* 0x02 */ s8  x;
-    /* 0x03 */ s8  y;
-} PadInput; // size = 0x4
+#define A_BUTTON        0x8000
+#define B_BUTTON        0x4000
+#define L_TRIG          0x0020
+#define R_TRIG          0x0010
+#define Z_TRIG          0x2000
+#define START_BUTTON    0x1000
+#define U_JPAD          0x0800
+#define L_JPAD          0x0200
+#define R_JPAD          0x0100
+#define D_JPAD          0x0400
```

Comment:
Id love to make JPAD DPAD instead, but idk where we stand on changing official names in libultra stuff. no comments from me codewise though, think it looks good